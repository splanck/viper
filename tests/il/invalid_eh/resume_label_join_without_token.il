il 0.1.2

func @resume_label_join_without_token(i1 %flag, Error %err, ResumeTok %tok) -> void {
entry:
  eh.push ^handler
  cbr %flag, handler(%err, %tok), trap_block

trap_block:
  trap

handler ^handler(%err:Error, %tok:ResumeTok):
  eh.entry
  resume.label %tok, ^after(%err)

after(%forward_err:Error):
  ret
}
