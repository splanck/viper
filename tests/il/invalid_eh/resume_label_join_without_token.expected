error: verify.eh.resume_token_missing: resume_label_join_without_token:handler: resume.label %t4 label after: [checkUnreachableAfterThrow] resume.* requires active resume token; path: entry -> handler
